src/seedu/addressbook/AddressBook.java

```diff
@@ -387,52 +400,61 @@ private static String getMessageForInvalidCommandInput(String userCommand, Strin
      */
     private static String executeAddPerson(String commandArgs) {
         // try decoding a person from the raw args
-        final Optional<String[]> decodeResult = decodePersonFromString(commandArgs);
+//        final Optional<String[]> decodeResult = decodePersonFromString(commandArgs);
+    	final Optional<HashMap<String, String>> decodeResult = decodePersonFromString(commandArgs);
 
         // checks if args are valid (decode result will not be present if the person is invalid)
         if (!decodeResult.isPresent()) {
             return getMessageForInvalidCommandInput(COMMAND_ADD_WORD, getUsageInfoForAddCommand());
         }
 
         // add the person as specified
-        final String[] personToAdd = decodeResult.get();
+        // Update: The original personToAdd is now renamed to personToAddStrArr. personToAdd is now a HashMap<String, String>.
+//        String[] personToAdd = decodeResult.get();
```

Comment:
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.